src/compiler/path.ts

```diff
@@ -720,10 +724,31 @@ export function getNormalizedAbsolutePath(path: string, currentDirectory: string
 
 /** @internal */
 export function normalizePath(path: string): string {
-    const normalized = getNormalizedAbsolutePath(path, "");
+    let normalized = simpleNormalizePath(path);
+    if (normalized !== undefined) {
+        return normalized;
+    }
+    normalized = getNormalizedAbsolutePath(path, "");
     return normalized && hasTrailingDirectorySeparator(path) ? ensureTrailingDirectorySeparator(normalized) : normalized;
 }
 
+function simpleNormalizePath(path: string): string | undefined {
+    path = normalizeSlashes(path);
+    // Most paths don't require normalization
+    if (!relativePathSegmentRegExp.test(path)) {
+        return path;
+    }
+    // Some paths only require cleanup of `/./` or leading `./`
+    const simplified = path.replace(/\/\.\//g, "/").replace(/^\.\//, "");

```

Comment:
If we're already here, I'd also recommend

```suggestion
    let simplified = path.replace(/\/\.\//g, "/");
    if (simplified.startsWith("./") {
      simplified = simplified.slice(2);
    }
```

Reply:
This is indeed a small improvement. Nice!